src/notifications/use-push-notifications.tsx

```diff
@@ -53,23 +55,34 @@ export const usePushNotifications = () => {
   }, []);
 
   const register = useCallback(async () => {
+    try {
+      const token = await messaging().getToken();
+      if (!token) return;
+      mutateRegister({token, language: language});
+      return token;
+    } catch (e) {
+      Bugsnag.notify(`Failed to register for push notifications: ${e}`);
+    }
+  }, [language, mutateRegister]);
+
+  const requestPermissions = useCallback(async () => {
     setStatus('updating');
     const permissionStatus = await requestUserPermission();
-    const token = await messaging().getToken();
+    const token = await register();
     if (!token) {
       setStatus('error');
       return;
     }
```

Comment:
Setting status `error` if no token should only be done when invoking `requestPermissions`, and not when doing just `register`?

Reply:
I've made the register function as fire-and-forget, since when I use it directly, it's not an issue if it fails once in a while, and it doesn't necessarily mean notifications aren't working. When we use `requestPermissions`, we should handle errors.

The functionality is maybe a bit specific for how it's used, but not sure how to improve it.